JoinedMessage.java

```diff
@@ -0,0 +1,13 @@
+// sent from one node in the mesh to all others
+// to tell them a new node has joined so they
+// can add the node to their participants list
+public class JoinedMessage extends MessageType
+{
+    public Participant newNode;
+
+    JoinedMessage(Participant sender,Participant newNode)
+    {
+        super(sender);
+        newNode = newNode;
```

Comment:
I detect that this code is problematic. According to the [Dodgy code (STYLE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#dodgy-code-style), [SA: Self assignment of local variable (SA_LOCAL_SELF_ASSIGNMENT)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#sa-self-assignment-of-local-variable-sa-local-self-assignment).
This method contains a self assignment of a local variable; e.g.
Such assignments are useless, and may indicate a logic error or typo.